### PR TITLE
Add My Image Garden 3.5.1

### PR DESCRIPTION
**Pull request by kosma**
_If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so._

After making all changes to the cask:
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:
- [x] Named the cask according to the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked that the cask was not already refused in [closed issues](https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed).

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Please give instructions as to how to get to the download.

**Comment by kosma** [comment]
> Please give instructions as to how to get to the download.

Unfortunately there's no direct "My Image Garden download" link anywhere I searched. The simplest way is to select a compatible product (for example PIXMA iP7270 inkjet printer) and proper category and OS ([direct link](http://search-asia.canon-asia.com/canon__asia_en__asia_p_en/search.x?q=&ie=utf8&cat=0&ct=Support&pagemax=10&imgsize=1&pdf=ok&zoom=1&hf=category%09zubaken&cf=model_sm%3APIXMA+iP7270&modelName=PIXMA+iP7270&ref=support-asia.canon-asia.com&pid=TtXpWBvkSEpGtWnzuF9H3w..&qid=yuIRB3WDN8afZhvnLpJvzRC7j2s5IqGL&d=DOWNLOADS%09macOS+v10.12), no idea how long it will keep working). I got the .dmg URL from this page: http://support-asia.canon-asia.com/contents/ASIA/EN/0200487310.html As far as my experience goes, there's only one version of My Image Garden for all the supported printers.

File: Casks/my-image-garden.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
 cask 'my-image-garden' do
-  version '3.5.1'
+  version '3.5.1,ea11'
   sha256 '03c4f7a4ea0e39063f2f51fc51826ea53a60159cd0f0542d37a45367dddc8131'
 
-  # c-wss.com was verified as official when first introduced to the cask
-  url 'http://gdlp01.c-wss.com/gds/3/0200004873/03/mmig-mac-3_5_1-ea11.dmg'
+  # gdlp01.c-wss.com was verified as official when first introduced to the cask
+  url "http://gdlp01.c-wss.com/gds/3/0200004873/03/mmig-mac-#{version.before_comma.dots_to_underscores}-#{version.after_comma}.dmg"
```
Comment (kosma) [inline Casks/my-image-garden.rb]: Note that the numbers in URL change every time a new version is released, so this is not upgrade-proof.

File: Casks/my-image-garden.rb [hunk 2]
```diff
@@ -1,13 +1,13 @@
 cask 'my-image-garden' do
-  version '3.5.1'
+  version '3.5.1,ea11'
   sha256 '03c4f7a4ea0e39063f2f51fc51826ea53a60159cd0f0542d37a45367dddc8131'
 
-  # c-wss.com was verified as official when first introduced to the cask
-  url 'http://gdlp01.c-wss.com/gds/3/0200004873/03/mmig-mac-3_5_1-ea11.dmg'
+  # gdlp01.c-wss.com was verified as official when first introduced to the cask
```
Comment (kosma) [inline Casks/my-image-garden.rb]: Are the any reasons not to include the whole `c-wss.com` domain? It's listed in whois as belonging to Canon Japan.

**Comment by kosma** [comment]
Tested the changes, everything looks good. What's the next step?

**Comment by vitorgalvao** [comment]
> What's the next step?

Merging. Which I’m doing now. Incorporated some of your comments and was just waiting for Travis to finish. Thank you for the contribution!

**Pull request merged**